### PR TITLE
Use url search params

### PR DESCRIPTION
**Pull request by axelssonHakan**
**Added URLSearchParams:**
http://localhost:1234/?app=drugcases

- hostname (default: localhost)
- port (default: 9076)
- app (required)

### CODE REVIEW

File: src/enigma/config.js [hunk 1]
```diff
@@ -1,14 +1,19 @@
-import schema from 'enigma.js/schemas/12.20.0.json';
+import schema from 'enigma.js/schemas/12.170.2.json';
 
 import listCache from './list-cache';
 import layoutCache from './layout-cache';
 
+const searchParams = new URLSearchParams(document.location.search);
```
Comment (peol) [inline src/enigma/config.js]: Hm, I'm thinking maybe we should support supplying the websocket URL directly instead? E.g.

```js
encodeURIComponent('wss://secure-host:443/app/foo')
"wss%3A%2F%2Fsecure-host%3A443%2Fapp%2Ffoo"
```

->

Our querystring: `?session=wss%3A%2F%2Fsecure-host%3A443%2Fapp%2Ffoo"`

File: src/enigma/config.js [hunk 2]
```diff
@@ -1,14 +1,19 @@
...
+const port = searchParams.get('port') || '9076';
+const app = searchParams.get('app') || 'enginedata';
+
 const ERR_ABORTED = 15;
-const url = process.env.ENGINE_URL;
 
 const config = {
   schema,
-  createSocket: () => new WebSocket(url || `ws://${location.hostname}:9076/app/${+new Date()}`), // eslint-disable-line
+  app,
```
Comment (peol) [inline src/enigma/config.js]: enigma.js does not have an `app` config property, it has a `url` one though. So in the suggestion above, the `?session=` value would be `url` here.

File: src/enigma/config.js [hunk 3]
```diff
@@ -1,14 +1,19 @@
...
+  createSocket: () => new WebSocket(`ws://${hostname}:${port}/app/engineData`),
```
Comment (peol) [inline src/enigma/config.js]: My bad, we should use the public enigma.js API, so `createSocket: url => new WebSocket(url)` (as long as the `config.url` value is set...)

**Comment by axelssonHakan** [comment]
New url Search Param (URIencoded ws URL)
/?session=ws%3A%2F%2Flocalhost%3A9076%2Fapp%2Fdrugcases

**Comment by axelssonHakan** [comment]
@snurrfint Your magic touch is needed on the error/input page
https://5bc6f32c1f12b71d629bcdbb--catwalk.netlify.com/

**Pull request merged**